pkg/webservice/handlers.go

```diff
@@ -898,6 +898,56 @@ func getPartitionRules(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getQueueApplicationsByState(w http.ResponseWriter, r *http.Request) {
+	writeHeaders(w)
+	vars := httprouter.ParamsFromContext(r.Context())
+	if vars == nil {
+		buildJSONErrorResponse(w, MissingParamsName, http.StatusBadRequest)
+		return
+	}
+
+	partition := vars.ByName("partition")
+	queueName := vars.ByName("queue")
+	appState := strings.ToLower(vars.ByName("state"))
+	status := strings.ToLower(r.URL.Query().Get("status"))
+
+	queueErr := validateQueue(queueName)
+	if queueErr != nil {
+		buildJSONErrorResponse(w, queueErr.Error(), http.StatusBadRequest)
+		return
+	}
+	partitionContext := schedulerContext.GetPartitionWithoutClusterID(partition)
```

Comment:
`schedulerContext.Load().GetPartitionWithoutClusterID(partition)`

Reply:
Done. Merged the latest master commit and made the adjustment.